File: src/challenge2.ts
class Stack<T> {
    private items: T[]

    constructor(){
        this.items = [];
    }

    push(value: T): void{
        this.items.push(value);
    }

    pop():T | undefined {
        return this.items.pop();
    }

    peek(): T | undefined {
        return this.items.length > 0 ? this.items[this.items.length -1]: undefined;
    }

    size(): number{
        return this.items.length;
    }
}

//test the imple
const s = new Stack<number>();
s.push(1);
s.push(2);
console.log(s.peek());  // Expected: 2
console.log(s.pop());    // Expected: 2
console.log(s.size());   // Expected: 1

File: challenge2.py
"""
Challenge 2: Stack/Queue Operations
"""

from stack import Stack


test_mode = False
s = Stack()

if not test_mode:
    '''
        The following code is taken directly from the challenge.
        Set `test_mode = False` to run this code.
    '''
    s.push(1)
    s.push(2)
    print(s.peek())  # Expected: 2
    print(s.pop())   # Expected: 2
    print(s.size())  # Expected: 1

else:
    print("Running in test mode...")
    print(s.size())
    # print(s.peek())  # Should crash
    # s.pop()  # Should crash

    print("Add 'dd'")
    s.push("dd")

    print("Remove 'dd'")
    print(s.pop())

    print("Add 'ee' and 'ff'")
    s.push("ee")
    s.push("ff")
    print(s.size())
    print(s.peek())

    print("Add 'gg'")
    s.push("gg")
    print(s.peek())

    print("Before delete:")
    print(s.data)
    print(s.size())
    print(s.pop())
    print("After delete:")
    print(s.peek())
    print(s.data)
    print(s.size())
